{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TEMPTED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gemelli, version 0.0.11\r\n"
     ]
    }
   ],
   "source": [
    "!gemelli --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved FeatureTable[Composition] to: /Users/callaband/Desktop/UCSD/KnightLab/Haddad/16S_HFHC_10wk_IHC/data/Microbiome/lr-metrics/tempted/clr-table.qza\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime gemelli clr-transformation \\\n",
    "    --i-table ../data/Microbiome/lr-metrics/longIHC_gg2filt_feces_table_10300.qza \\\n",
    "    --o-clr-table ../data/Microbiome/lr-metrics/tempted/clr-table.qza\n",
    "#does not include samples that would have been rarified out, but non-rarified; otherwise contains all samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mExported /Users/callaband/Desktop/UCSD/KnightLab/Haddad/16S_HFHC_10wk_IHC/data/Microbiome/lr-metrics/tempted/clr-table.qza as BIOMV210DirFmt to directory /Users/callaband/Desktop/UCSD/KnightLab/Haddad/16S_HFHC_10wk_IHC/data/Microbiome/lr-metrics/tempted/clr-table\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime tools export \\\n",
    "    --input-path ../data/Microbiome/lr-metrics/tempted/clr-table.qza \\\n",
    "    --output-path ../data/Microbiome/lr-metrics/tempted/clr-table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "!biom convert -i ../data/Microbiome/lr-metrics/tempted/clr-table/feature-table.biom -o ../data/Microbiome/lr-metrics/tempted/clr-table/feature-table.tsv --to-tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>11548.A16.91.4</th>\n",
       "      <th>11548.A15.85.11</th>\n",
       "      <th>11548.A15.87.8</th>\n",
       "      <th>11548.A16.92.3</th>\n",
       "      <th>11548.A16.89.6</th>\n",
       "      <th>11548.A15.86.9</th>\n",
       "      <th>11548.A15.88.7</th>\n",
       "      <th>11548.A17.93.20</th>\n",
       "      <th>11548.A16.89.7</th>\n",
       "      <th>11548.A16.92.4</th>\n",
       "      <th>...</th>\n",
       "      <th>11548.A23.120.2</th>\n",
       "      <th>11548.A23.117.7</th>\n",
       "      <th>11548.A22.115.9</th>\n",
       "      <th>11548.A23.119.3</th>\n",
       "      <th>11548.A23.118.6</th>\n",
       "      <th>11548.A23.119.4</th>\n",
       "      <th>11548.A23.120.20</th>\n",
       "      <th>11548.A23.120.3</th>\n",
       "      <th>11548.A23.119.5</th>\n",
       "      <th>11548.A23.120.4</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#OTU ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGACTGTCAAGTCAGCGGTAAAATTGAGAGGCTCAACCTCTTCCCGCCGTTGAAACTGGTGGTCTTGAGTGGATGAGAAGTACGCGGAATGCGTGGT</th>\n",
       "      <td>8.291201</td>\n",
       "      <td>7.541094</td>\n",
       "      <td>7.709825</td>\n",
       "      <td>7.426568</td>\n",
       "      <td>8.362102</td>\n",
       "      <td>7.840682</td>\n",
       "      <td>8.096147</td>\n",
       "      <td>6.450197</td>\n",
       "      <td>7.680043</td>\n",
       "      <td>8.253643</td>\n",
       "      <td>...</td>\n",
       "      <td>5.535091</td>\n",
       "      <td>4.833216</td>\n",
       "      <td>6.076420</td>\n",
       "      <td>3.507330</td>\n",
       "      <td>5.680663</td>\n",
       "      <td>3.890926</td>\n",
       "      <td>5.801532</td>\n",
       "      <td>4.084202</td>\n",
       "      <td>3.660403</td>\n",
       "      <td>4.857618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGTCCGTTAAGTCAGCGGTAAAATTGCGGGGCTCAACCCCGTCGAGCCGTTGAAACTGGCAGACTTGAGTTGGCGAGAAGTACGCGGAATGCGCGGT</th>\n",
       "      <td>7.326821</td>\n",
       "      <td>7.430599</td>\n",
       "      <td>7.863914</td>\n",
       "      <td>6.121906</td>\n",
       "      <td>7.256140</td>\n",
       "      <td>8.941068</td>\n",
       "      <td>8.444831</td>\n",
       "      <td>7.840313</td>\n",
       "      <td>7.433183</td>\n",
       "      <td>7.102850</td>\n",
       "      <td>...</td>\n",
       "      <td>5.874829</td>\n",
       "      <td>5.900329</td>\n",
       "      <td>7.333002</td>\n",
       "      <td>4.634930</td>\n",
       "      <td>5.905130</td>\n",
       "      <td>5.114702</td>\n",
       "      <td>7.808387</td>\n",
       "      <td>5.220352</td>\n",
       "      <td>5.616970</td>\n",
       "      <td>5.998264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGTCGGATAAGTTAGAGGTGAAATCCCGAGGCTCAACTTCGGAATTGCCTCTGATACTGTTCGGCTAGAGAGTAGTTGCGGTAGGCGGAATGTATGG</th>\n",
       "      <td>7.197212</td>\n",
       "      <td>6.729905</td>\n",
       "      <td>6.547799</td>\n",
       "      <td>6.984224</td>\n",
       "      <td>6.347771</td>\n",
       "      <td>7.126393</td>\n",
       "      <td>6.573234</td>\n",
       "      <td>6.023059</td>\n",
       "      <td>6.831703</td>\n",
       "      <td>7.255294</td>\n",
       "      <td>...</td>\n",
       "      <td>2.799959</td>\n",
       "      <td>3.192279</td>\n",
       "      <td>3.609212</td>\n",
       "      <td>2.545919</td>\n",
       "      <td>3.995587</td>\n",
       "      <td>3.444639</td>\n",
       "      <td>3.920893</td>\n",
       "      <td>2.917767</td>\n",
       "      <td>4.535871</td>\n",
       "      <td>2.459723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGCAGCACAAGTCTGGAGTGAAATCCCGGGGCTTAACCCCGGAACTGCTTTGGAAACTGTGCGGCTGGAGTGCAGGAGGGGTAAGCGGAATTCCTAG</th>\n",
       "      <td>6.787002</td>\n",
       "      <td>-0.024699</td>\n",
       "      <td>-0.035610</td>\n",
       "      <td>7.997937</td>\n",
       "      <td>5.769544</td>\n",
       "      <td>-0.033676</td>\n",
       "      <td>-0.044168</td>\n",
       "      <td>1.578058</td>\n",
       "      <td>7.423998</td>\n",
       "      <td>6.832258</td>\n",
       "      <td>...</td>\n",
       "      <td>1.576184</td>\n",
       "      <td>-0.026597</td>\n",
       "      <td>-0.028374</td>\n",
       "      <td>-0.019030</td>\n",
       "      <td>2.167460</td>\n",
       "      <td>-0.021097</td>\n",
       "      <td>-0.030351</td>\n",
       "      <td>-0.026672</td>\n",
       "      <td>0.664670</td>\n",
       "      <td>-0.025184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGGATGCCAAGTCAGCGGTAAAAAAGCGGTGCTCAACGCCGTCGAGCCGTTGAAACTGGCGTTCTTGAGTGGGCGAGAAGTATGCGGAATGCGTGGT</th>\n",
       "      <td>6.714160</td>\n",
       "      <td>7.818365</td>\n",
       "      <td>7.338391</td>\n",
       "      <td>6.930952</td>\n",
       "      <td>7.540705</td>\n",
       "      <td>7.682339</td>\n",
       "      <td>6.914280</td>\n",
       "      <td>7.701866</td>\n",
       "      <td>6.949602</td>\n",
       "      <td>7.361474</td>\n",
       "      <td>...</td>\n",
       "      <td>4.648877</td>\n",
       "      <td>4.848601</td>\n",
       "      <td>4.799940</td>\n",
       "      <td>3.618556</td>\n",
       "      <td>5.583364</td>\n",
       "      <td>4.213010</td>\n",
       "      <td>5.069516</td>\n",
       "      <td>3.943620</td>\n",
       "      <td>4.130406</td>\n",
       "      <td>4.842350</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 690 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    11548.A16.91.4  \\\n",
       "#OTU ID                                                              \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        8.291201   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.326821   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.197212   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        6.787002   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        6.714160   \n",
       "\n",
       "                                                    11548.A15.85.11  \\\n",
       "#OTU ID                                                               \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         7.541094   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         7.430599   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         6.729905   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        -0.024699   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         7.818365   \n",
       "\n",
       "                                                    11548.A15.87.8  \\\n",
       "#OTU ID                                                              \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.709825   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.863914   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        6.547799   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...       -0.035610   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.338391   \n",
       "\n",
       "                                                    11548.A16.92.3  \\\n",
       "#OTU ID                                                              \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.426568   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        6.121906   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        6.984224   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        7.997937   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        6.930952   \n",
       "\n",
       "                                                    11548.A16.89.6  \\\n",
       "#OTU ID                                                              \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        8.362102   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.256140   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        6.347771   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        5.769544   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.540705   \n",
       "\n",
       "                                                    11548.A15.86.9  \\\n",
       "#OTU ID                                                              \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.840682   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        8.941068   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.126393   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...       -0.033676   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.682339   \n",
       "\n",
       "                                                    11548.A15.88.7  \\\n",
       "#OTU ID                                                              \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        8.096147   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        8.444831   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        6.573234   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...       -0.044168   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        6.914280   \n",
       "\n",
       "                                                    11548.A17.93.20  \\\n",
       "#OTU ID                                                               \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         6.450197   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         7.840313   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         6.023059   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...         1.578058   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         7.701866   \n",
       "\n",
       "                                                    11548.A16.89.7  \\\n",
       "#OTU ID                                                              \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.680043   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.433183   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        6.831703   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        7.423998   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        6.949602   \n",
       "\n",
       "                                                    11548.A16.92.4  ...  \\\n",
       "#OTU ID                                                             ...   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        8.253643  ...   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.102850  ...   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.255294  ...   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        6.832258  ...   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...        7.361474  ...   \n",
       "\n",
       "                                                    11548.A23.120.2  \\\n",
       "#OTU ID                                                               \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         5.535091   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         5.874829   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         2.799959   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...         1.576184   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         4.648877   \n",
       "\n",
       "                                                    11548.A23.117.7  \\\n",
       "#OTU ID                                                               \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         4.833216   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         5.900329   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         3.192279   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        -0.026597   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         4.848601   \n",
       "\n",
       "                                                    11548.A22.115.9  \\\n",
       "#OTU ID                                                               \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         6.076420   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         7.333002   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         3.609212   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        -0.028374   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         4.799940   \n",
       "\n",
       "                                                    11548.A23.119.3  \\\n",
       "#OTU ID                                                               \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         3.507330   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         4.634930   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         2.545919   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        -0.019030   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         3.618556   \n",
       "\n",
       "                                                    11548.A23.118.6  \\\n",
       "#OTU ID                                                               \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         5.680663   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         5.905130   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         3.995587   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...         2.167460   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         5.583364   \n",
       "\n",
       "                                                    11548.A23.119.4  \\\n",
       "#OTU ID                                                               \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         3.890926   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         5.114702   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         3.444639   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        -0.021097   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         4.213010   \n",
       "\n",
       "                                                    11548.A23.120.20  \\\n",
       "#OTU ID                                                                \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...          5.801532   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...          7.808387   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...          3.920893   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...         -0.030351   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...          5.069516   \n",
       "\n",
       "                                                    11548.A23.120.3  \\\n",
       "#OTU ID                                                               \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         4.084202   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         5.220352   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         2.917767   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        -0.026672   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         3.943620   \n",
       "\n",
       "                                                    11548.A23.119.5  \\\n",
       "#OTU ID                                                               \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         3.660403   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         5.616970   \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         4.535871   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...         0.664670   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         4.130406   \n",
       "\n",
       "                                                    11548.A23.120.4  \n",
       "#OTU ID                                                              \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         4.857618  \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         5.998264  \n",
       "TACGGAGGATTCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         2.459723  \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...        -0.025184  \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...         4.842350  \n",
       "\n",
       "[5 rows x 690 columns]"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clr = pd.read_csv('../data/Microbiome/lr-metrics/tempted/clr-table/feature-table.tsv', \n",
    "                  skiprows=1, sep='\\t',index_col=0)\n",
    "clr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clr.isnull().values.any()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "qiime gemelli tempted-factorize \\\n",
    "    --i-table ../data/Microbiome/lr-metrics/tempted/clr-table.qza \\\n",
    "    --m-sample-metadata-file ../data/Microbiome/longIHC_metadata_cleaned.txt \\\n",
    "    --p-individual-id-column 'host_subject_id' \\\n",
    "    --p-state-column 'host_age' \\\n",
    "    --o-individual-biplot ../data/Microbiome/lr-metrics/tempted/individual_biplot.qza \\\n",
    "    --o-state-loadings ../data/Microbiome/lr-metrics/tempted/state_loadings.qza \\\n",
    "    --o-distance-matrix ../data/Microbiome/lr-metrics/tempted/distance_matrix.qza \\\n",
    "    --o-svd-center ../data/Microbiome/lr-metrics/tempted/svd_center.qza\n",
    "#Make sure to use gemelli version 0.0.12+ (fixed bug)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>diet_exp</th>\n",
       "      <th>diet</th>\n",
       "      <th>exposure_type</th>\n",
       "      <th>cage_number</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Mouse 100</th>\n",
       "      <td>HFD_IHC</td>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>IHC</td>\n",
       "      <td>A18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mouse 101</th>\n",
       "      <td>HFD_IHC</td>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>IHC</td>\n",
       "      <td>A19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mouse 102</th>\n",
       "      <td>HFD_IHC</td>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>IHC</td>\n",
       "      <td>A19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mouse 103</th>\n",
       "      <td>HFD_IHC</td>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>IHC</td>\n",
       "      <td>A19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mouse 104</th>\n",
       "      <td>HFD_IHC</td>\n",
       "      <td>High Fat Diet</td>\n",
       "      <td>IHC</td>\n",
       "      <td>A19</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          diet_exp           diet exposure_type cage_number\n",
       "#SampleID                                                  \n",
       "Mouse 100  HFD_IHC  High Fat Diet           IHC         A18\n",
       "Mouse 101  HFD_IHC  High Fat Diet           IHC         A19\n",
       "Mouse 102  HFD_IHC  High Fat Diet           IHC         A19\n",
       "Mouse 103  HFD_IHC  High Fat Diet           IHC         A19\n",
       "Mouse 104  HFD_IHC  High Fat Diet           IHC         A19"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# first we import the metdata into pandas\n",
    "mf = pd.read_csv('../data/Microbiome/longIHC_metadata_cleaned.txt', sep='\\t',index_col=0)\n",
    "# next we aggregate by subjects (i.e. 'host_subject_id') \n",
    "mf = mf.groupby('host_subject_id').agg({'diet_exp':'first','diet':'first', 'exposure_type':'first', 'cage_number':'first'})\n",
    "# now we save the metadata in QIIME2 format.\n",
    "mf.index.name = '#SampleID'\n",
    "mf.index = mf.index.astype(str)\n",
    "mf.to_csv('../data/Microbiome/lr-metrics/tempted/tempted-metadata.tsv', sep='\\t')\n",
    "mf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: /Users/callaband/Desktop/UCSD/KnightLab/Haddad/16S_HFHC_10wk_IHC/data/Microbiome/lr-metrics/tempted/subject_biplot.qzv\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime emperor biplot \\\n",
    "    --i-biplot ../data/Microbiome/lr-metrics/tempted/individual_biplot.qza \\\n",
    "    --m-sample-metadata-file ../data/Microbiome/lr-metrics/tempted/tempted-metadata.tsv \\\n",
    "    --m-feature-metadata-file ../data/Microbiome/longIHC_gg2filt_taxonomy.qza \\\n",
    "    --p-number-of-features 10\\\n",
    "    --o-visualization ../data/Microbiome/lr-metrics/tempted/subject_biplot.qzv\n",
    "#Looks great! Separates the three groups well!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: /Users/callaband/Desktop/UCSD/KnightLab/Haddad/16S_HFHC_10wk_IHC/data/Microbiome/lr-metrics/tempted/tempted_qurro.qzv\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime qurro loading-plot \\\n",
    "    --i-table ../data/Microbiome/lr-metrics/longIHC_gg2filt_feces_table_10300.qza \\\n",
    "    --i-ranks ../data/Microbiome/lr-metrics/tempted/individual_biplot.qza \\\n",
    "    --m-sample-metadata-file ../data/Microbiome/longIHC_metadata_cleaned.txt \\\n",
    "    --m-feature-metadata-file ../data/Microbiome/longIHC_gg2filt_taxonomy.qza \\\n",
    "    --o-visualization ../data/Microbiome/lr-metrics/tempted/tempted_qurro.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TopBot10Num_Axis1</th>\n",
       "      <th>anonymized_name</th>\n",
       "      <th>cage_food_consumption_g</th>\n",
       "      <th>cage_location</th>\n",
       "      <th>cage_notes</th>\n",
       "      <th>cage_number</th>\n",
       "      <th>collection_timestamp</th>\n",
       "      <th>description</th>\n",
       "      <th>diet</th>\n",
       "      <th>dna_extracted</th>\n",
       "      <th>...</th>\n",
       "      <th>sample_name_orit</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>sex</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>title</th>\n",
       "      <th>treatment</th>\n",
       "      <th>diet_short</th>\n",
       "      <th>diet_exp</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11548.A16.91.4</th>\n",
       "      <td>3.071426</td>\n",
       "      <td>A16.91.4</td>\n",
       "      <td>61</td>\n",
       "      <td>on shelf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A16</td>\n",
       "      <td>10/6/17 11:00</td>\n",
       "      <td>Mouse 91 stool collection 4 of 20</td>\n",
       "      <td>Regular chow</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>a16.91.4</td>\n",
       "      <td>feces</td>\n",
       "      <td>mouse gut metagenome</td>\n",
       "      <td>male</td>\n",
       "      <td>410661</td>\n",
       "      <td>not provided</td>\n",
       "      <td>10 week IHC</td>\n",
       "      <td>not provided</td>\n",
       "      <td>RC</td>\n",
       "      <td>RC_AIR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11548.A15.85.11</th>\n",
       "      <td>1.887561</td>\n",
       "      <td>A15.85.11</td>\n",
       "      <td>96.7</td>\n",
       "      <td>on shelf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A15</td>\n",
       "      <td>10/31/17 11:00</td>\n",
       "      <td>Mouse 85 stool collection 11 of 20</td>\n",
       "      <td>Regular chow</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>a15.85.11</td>\n",
       "      <td>feces</td>\n",
       "      <td>mouse gut metagenome</td>\n",
       "      <td>male</td>\n",
       "      <td>410661</td>\n",
       "      <td>not provided</td>\n",
       "      <td>10 week IHC</td>\n",
       "      <td>not provided</td>\n",
       "      <td>RC</td>\n",
       "      <td>RC_AIR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11548.A15.87.8</th>\n",
       "      <td>3.035758</td>\n",
       "      <td>A15.87.8</td>\n",
       "      <td>75.9</td>\n",
       "      <td>on shelf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A15</td>\n",
       "      <td>10/20/17 11:00</td>\n",
       "      <td>Mouse 87 stool collection 8 of 20</td>\n",
       "      <td>Regular chow</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>a15.87.8</td>\n",
       "      <td>feces</td>\n",
       "      <td>mouse gut metagenome</td>\n",
       "      <td>male</td>\n",
       "      <td>410661</td>\n",
       "      <td>not provided</td>\n",
       "      <td>10 week IHC</td>\n",
       "      <td>not provided</td>\n",
       "      <td>RC</td>\n",
       "      <td>RC_AIR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11548.A16.92.3</th>\n",
       "      <td>3.559083</td>\n",
       "      <td>A16.92.3</td>\n",
       "      <td>85.4</td>\n",
       "      <td>on shelf</td>\n",
       "      <td>put nyla bones in cage</td>\n",
       "      <td>A16</td>\n",
       "      <td>10/3/17 11:00</td>\n",
       "      <td>Mouse 92 stool collection 3 of 20</td>\n",
       "      <td>Regular chow</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>a16.92.3</td>\n",
       "      <td>feces</td>\n",
       "      <td>mouse gut metagenome</td>\n",
       "      <td>male</td>\n",
       "      <td>410661</td>\n",
       "      <td>not provided</td>\n",
       "      <td>10 week IHC</td>\n",
       "      <td>not provided</td>\n",
       "      <td>RC</td>\n",
       "      <td>RC_AIR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11548.A16.89.6</th>\n",
       "      <td>-0.271649</td>\n",
       "      <td>A16.89.6</td>\n",
       "      <td>65.3</td>\n",
       "      <td>on shelf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A16</td>\n",
       "      <td>10/13/17 11:00</td>\n",
       "      <td>Mouse 89 stool collection 6 of 20</td>\n",
       "      <td>Regular chow</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>A16.89.6</td>\n",
       "      <td>feces</td>\n",
       "      <td>mouse gut metagenome</td>\n",
       "      <td>male</td>\n",
       "      <td>410661</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10 week IHC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RC</td>\n",
       "      <td>RC_AIR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 TopBot10Num_Axis1 anonymized_name cage_food_consumption_g  \\\n",
       "#SampleID                                                                    \n",
       "11548.A16.91.4            3.071426        A16.91.4                      61   \n",
       "11548.A15.85.11           1.887561       A15.85.11                    96.7   \n",
       "11548.A15.87.8            3.035758        A15.87.8                    75.9   \n",
       "11548.A16.92.3            3.559083        A16.92.3                    85.4   \n",
       "11548.A16.89.6           -0.271649        A16.89.6                    65.3   \n",
       "\n",
       "                cage_location              cage_notes cage_number  \\\n",
       "#SampleID                                                           \n",
       "11548.A16.91.4       on shelf                     NaN         A16   \n",
       "11548.A15.85.11      on shelf                     NaN         A15   \n",
       "11548.A15.87.8       on shelf                     NaN         A15   \n",
       "11548.A16.92.3       on shelf  put nyla bones in cage         A16   \n",
       "11548.A16.89.6       on shelf                     NaN         A16   \n",
       "\n",
       "                collection_timestamp                         description  \\\n",
       "#SampleID                                                                  \n",
       "11548.A16.91.4         10/6/17 11:00   Mouse 91 stool collection 4 of 20   \n",
       "11548.A15.85.11       10/31/17 11:00  Mouse 85 stool collection 11 of 20   \n",
       "11548.A15.87.8        10/20/17 11:00   Mouse 87 stool collection 8 of 20   \n",
       "11548.A16.92.3         10/3/17 11:00   Mouse 92 stool collection 3 of 20   \n",
       "11548.A16.89.6        10/13/17 11:00   Mouse 89 stool collection 6 of 20   \n",
       "\n",
       "                         diet  dna_extracted  ... sample_name_orit  \\\n",
       "#SampleID                                     ...                    \n",
       "11548.A16.91.4   Regular chow           True  ...         a16.91.4   \n",
       "11548.A15.85.11  Regular chow           True  ...        a15.85.11   \n",
       "11548.A15.87.8   Regular chow           True  ...         a15.87.8   \n",
       "11548.A16.92.3   Regular chow           True  ...         a16.92.3   \n",
       "11548.A16.89.6   Regular chow           True  ...         A16.89.6   \n",
       "\n",
       "                 sample_type       scientific_name   sex taxon_id  \\\n",
       "#SampleID                                                           \n",
       "11548.A16.91.4         feces  mouse gut metagenome  male   410661   \n",
       "11548.A15.85.11        feces  mouse gut metagenome  male   410661   \n",
       "11548.A15.87.8         feces  mouse gut metagenome  male   410661   \n",
       "11548.A16.92.3         feces  mouse gut metagenome  male   410661   \n",
       "11548.A16.89.6         feces  mouse gut metagenome  male   410661   \n",
       "\n",
       "                    timepoint        title     treatment diet_short diet_exp  \n",
       "#SampleID                                                                     \n",
       "11548.A16.91.4   not provided  10 week IHC  not provided         RC   RC_AIR  \n",
       "11548.A15.85.11  not provided  10 week IHC  not provided         RC   RC_AIR  \n",
       "11548.A15.87.8   not provided  10 week IHC  not provided         RC   RC_AIR  \n",
       "11548.A16.92.3   not provided  10 week IHC  not provided         RC   RC_AIR  \n",
       "11548.A16.89.6            NaN  10 week IHC           NaN         RC   RC_AIR  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# import log-ratio data\n",
    "metadata_one = pd.read_csv('../data/Microbiome/longIHC_metadata_cleaned.txt',\n",
    "                           sep='\\t', index_col=0)\n",
    "# import rest of the metadata\n",
    "metadata_two = pd.read_csv('../data/Microbiome/lr-metrics/tempted/longIHC_tempted_topbot10num_Axis1_lr.tsv',\n",
    "                           sep='\\t', index_col=0)[['Current_Natural_Log_Ratio']]\n",
    "#rename LR column so you know what they are\n",
    "metadata_two.rename(columns={'Current_Natural_Log_Ratio':'TopBot10Num_Axis1'}, inplace=True)\n",
    "# merge the data\n",
    "log_ratio_metdata = pd.concat([metadata_two, metadata_one], axis=1)\n",
    "#drop rows with no value for LR\n",
    "log_ratio_metdata = log_ratio_metdata.dropna(subset=['TopBot10Num_Axis1'])\n",
    "# export in QIIME2 format\n",
    "log_ratio_metdata.index.name = '#SampleID'\n",
    "log_ratio_metdata.to_csv('../data/Microbiome/lr-metrics/tempted/lr_merged-data.tsv', sep='\\t')\n",
    "log_ratio_metdata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../data/Microbiome/lr-metrics/tempted/log_ratio_plot.qzv\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime longitudinal volatility \\\n",
    "    --m-metadata-file ../data/Microbiome/lr-metrics/tempted/lr_merged-data.tsv \\\n",
    "    --p-state-column host_age \\\n",
    "    --p-individual-id-column host_subject_id \\\n",
    "    --p-default-group-column diet_exp \\\n",
    "    --p-default-metric TopBot10Num_Axis1 \\\n",
    "    --o-visualization ../data/Microbiome/lr-metrics/tempted/log_ratio_plot.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../data/Microbiome/lr-metrics/tempted/lme_log_ratio.qzv\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime longitudinal linear-mixed-effects\\\n",
    "    --m-metadata-file ../data/Microbiome/lr-metrics/tempted/lr_merged-data.tsv \\\n",
    "    --p-state-column host_age \\\n",
    "    --p-individual-id-column host_subject_id \\\n",
    "    --p-group-columns diet_exp \\\n",
    "    --p-metric TopBot10Num_Axis1 \\\n",
    "    --o-visualization ../data/Microbiome/lr-metrics/tempted/lme_log_ratio.qzv\n",
    "#Looks great, significant for all comparisons!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "topfeat = pd.read_csv('../data/Microbiome/lr-metrics/tempted/longIHC_tempted_topbot10num_Axis1_lr_features.tsv',\n",
    "                     sep='\\t', index_col=0)\n",
    "taxonomy = pd.read_csv('../data/Microbiome/longIHC_gg2filt_taxonomy/taxonomy.tsv', sep='\\t', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Log_Ratio_Classification</th>\n",
       "      <th>Taxon</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Feature ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGTAATGTAAGTCTGATGTGAAAGCCCGGGGCTCAACCCCGGGACTGCATTGGAAACTATGTAACTAGAGTACCGGAGAGGTAAGCGGAATTCCTAG</th>\n",
       "      <td>Denominator</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_A; c__Clostridia_25...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TACGTAGGTGGCGAGCGTTATCCGGAATCATTGGGCGTAAAGAGGGAGCAGGCGGCCTGGAGGGTCTATTGTTAAAAGGCAGTGGCTTAACCATTGCAAGGCGAAGAAACCGGCAGGCTAGAGTTCTAAAGAGGATCGTGGAATTCCATG</th>\n",
       "      <td>Denominator</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_D; c__Bacilli; o__E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TACGTATGGTGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGCGTAGACGGAGAGGCAAGTCTGATGTGAAAACCCGGGGCTCAACCCCGGGACTGCATTGGAAACTGTTTTTCTAGAGTGTCGGAGAGGCAAGTGGAATTCCTAG</th>\n",
       "      <td>Denominator</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_A; c__Clostridia_25...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TACGTAGGGGGCAAGCGTTATCCGGAATTACTGGGTGTAAAGGGAGCGTAGACGGTGATGCAAGTCTGGAGTGAAAGGCGGGGGCCCAACCCCCGGACTGCTCTGGAAACTGTGTAACTTGAGTGCAGGAGAGGTAAGCGGAATTCCTAG</th>\n",
       "      <td>Denominator</td>\n",
       "      <td>d__Bacteria; p__Firmicutes_A; c__Clostridia_25...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGCAGGCGGAAGCGCAAGTCAGCGGTAAAATTGAGAGGCTCAACCTCTTCCCGCCGTTGAAACTGCGTTTCTTGAGTGGGCGAGAAGTACGCGGAATGCGTGGT</th>\n",
       "      <td>Denominator</td>\n",
       "      <td>d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Log_Ratio_Classification  \\\n",
       "Feature ID                                                                    \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...              Denominator   \n",
       "TACGTAGGTGGCGAGCGTTATCCGGAATCATTGGGCGTAAAGAGGGA...              Denominator   \n",
       "TACGTATGGTGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...              Denominator   \n",
       "TACGTAGGGGGCAAGCGTTATCCGGAATTACTGGGTGTAAAGGGAGC...              Denominator   \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...              Denominator   \n",
       "\n",
       "                                                                                                Taxon  \n",
       "Feature ID                                                                                             \n",
       "TACGTAGGGGGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...  d__Bacteria; p__Firmicutes_A; c__Clostridia_25...  \n",
       "TACGTAGGTGGCGAGCGTTATCCGGAATCATTGGGCGTAAAGAGGGA...  d__Bacteria; p__Firmicutes_D; c__Bacilli; o__E...  \n",
       "TACGTATGGTGCAAGCGTTATCCGGATTTACTGGGTGTAAAGGGAGC...  d__Bacteria; p__Firmicutes_A; c__Clostridia_25...  \n",
       "TACGTAGGGGGCAAGCGTTATCCGGAATTACTGGGTGTAAAGGGAGC...  d__Bacteria; p__Firmicutes_A; c__Clostridia_25...  \n",
       "TACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGC...  d__Bacteria; p__Bacteroidota; c__Bacteroidia; ...  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "topfeat_named = pd.merge(topfeat, taxonomy[['Taxon']], left_index=True, right_index=True)\n",
    "topfeat_named.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "topfeat_named.to_csv('../data/Microbiome/lr-metrics/tempted/longIHC_tempted_topbot10num_Axis1_lr_features_named.tsv',\n",
    "                     sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Muribaculaceae_Akkermansiaceae</th>\n",
       "      <th>anonymized_name</th>\n",
       "      <th>cage_food_consumption_g</th>\n",
       "      <th>cage_location</th>\n",
       "      <th>cage_notes</th>\n",
       "      <th>cage_number</th>\n",
       "      <th>collection_timestamp</th>\n",
       "      <th>description</th>\n",
       "      <th>diet</th>\n",
       "      <th>dna_extracted</th>\n",
       "      <th>...</th>\n",
       "      <th>sample_name_orit</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>sex</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>timepoint</th>\n",
       "      <th>title</th>\n",
       "      <th>treatment</th>\n",
       "      <th>diet_short</th>\n",
       "      <th>diet_exp</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11548.A16.91.4</th>\n",
       "      <td>3.512776</td>\n",
       "      <td>A16.91.4</td>\n",
       "      <td>61</td>\n",
       "      <td>on shelf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A16</td>\n",
       "      <td>10/6/17 11:00</td>\n",
       "      <td>Mouse 91 stool collection 4 of 20</td>\n",
       "      <td>Regular chow</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>a16.91.4</td>\n",
       "      <td>feces</td>\n",
       "      <td>mouse gut metagenome</td>\n",
       "      <td>male</td>\n",
       "      <td>410661</td>\n",
       "      <td>not provided</td>\n",
       "      <td>10 week IHC</td>\n",
       "      <td>not provided</td>\n",
       "      <td>RC</td>\n",
       "      <td>RC_AIR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11548.A15.85.11</th>\n",
       "      <td>2.237845</td>\n",
       "      <td>A15.85.11</td>\n",
       "      <td>96.7</td>\n",
       "      <td>on shelf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A15</td>\n",
       "      <td>10/31/17 11:00</td>\n",
       "      <td>Mouse 85 stool collection 11 of 20</td>\n",
       "      <td>Regular chow</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>a15.85.11</td>\n",
       "      <td>feces</td>\n",
       "      <td>mouse gut metagenome</td>\n",
       "      <td>male</td>\n",
       "      <td>410661</td>\n",
       "      <td>not provided</td>\n",
       "      <td>10 week IHC</td>\n",
       "      <td>not provided</td>\n",
       "      <td>RC</td>\n",
       "      <td>RC_AIR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11548.A15.87.8</th>\n",
       "      <td>4.231688</td>\n",
       "      <td>A15.87.8</td>\n",
       "      <td>75.9</td>\n",
       "      <td>on shelf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A15</td>\n",
       "      <td>10/20/17 11:00</td>\n",
       "      <td>Mouse 87 stool collection 8 of 20</td>\n",
       "      <td>Regular chow</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>a15.87.8</td>\n",
       "      <td>feces</td>\n",
       "      <td>mouse gut metagenome</td>\n",
       "      <td>male</td>\n",
       "      <td>410661</td>\n",
       "      <td>not provided</td>\n",
       "      <td>10 week IHC</td>\n",
       "      <td>not provided</td>\n",
       "      <td>RC</td>\n",
       "      <td>RC_AIR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11548.A16.92.3</th>\n",
       "      <td>3.950938</td>\n",
       "      <td>A16.92.3</td>\n",
       "      <td>85.4</td>\n",
       "      <td>on shelf</td>\n",
       "      <td>put nyla bones in cage</td>\n",
       "      <td>A16</td>\n",
       "      <td>10/3/17 11:00</td>\n",
       "      <td>Mouse 92 stool collection 3 of 20</td>\n",
       "      <td>Regular chow</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>a16.92.3</td>\n",
       "      <td>feces</td>\n",
       "      <td>mouse gut metagenome</td>\n",
       "      <td>male</td>\n",
       "      <td>410661</td>\n",
       "      <td>not provided</td>\n",
       "      <td>10 week IHC</td>\n",
       "      <td>not provided</td>\n",
       "      <td>RC</td>\n",
       "      <td>RC_AIR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11548.A16.89.6</th>\n",
       "      <td>-0.121176</td>\n",
       "      <td>A16.89.6</td>\n",
       "      <td>65.3</td>\n",
       "      <td>on shelf</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A16</td>\n",
       "      <td>10/13/17 11:00</td>\n",
       "      <td>Mouse 89 stool collection 6 of 20</td>\n",
       "      <td>Regular chow</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>A16.89.6</td>\n",
       "      <td>feces</td>\n",
       "      <td>mouse gut metagenome</td>\n",
       "      <td>male</td>\n",
       "      <td>410661</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10 week IHC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RC</td>\n",
       "      <td>RC_AIR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 Muribaculaceae_Akkermansiaceae anonymized_name  \\\n",
       "#SampleID                                                         \n",
       "11548.A16.91.4                         3.512776        A16.91.4   \n",
       "11548.A15.85.11                        2.237845       A15.85.11   \n",
       "11548.A15.87.8                         4.231688        A15.87.8   \n",
       "11548.A16.92.3                         3.950938        A16.92.3   \n",
       "11548.A16.89.6                        -0.121176        A16.89.6   \n",
       "\n",
       "                cage_food_consumption_g cage_location              cage_notes  \\\n",
       "#SampleID                                                                       \n",
       "11548.A16.91.4                       61      on shelf                     NaN   \n",
       "11548.A15.85.11                    96.7      on shelf                     NaN   \n",
       "11548.A15.87.8                     75.9      on shelf                     NaN   \n",
       "11548.A16.92.3                     85.4      on shelf  put nyla bones in cage   \n",
       "11548.A16.89.6                     65.3      on shelf                     NaN   \n",
       "\n",
       "                cage_number collection_timestamp  \\\n",
       "#SampleID                                          \n",
       "11548.A16.91.4          A16        10/6/17 11:00   \n",
       "11548.A15.85.11         A15       10/31/17 11:00   \n",
       "11548.A15.87.8          A15       10/20/17 11:00   \n",
       "11548.A16.92.3          A16        10/3/17 11:00   \n",
       "11548.A16.89.6          A16       10/13/17 11:00   \n",
       "\n",
       "                                        description          diet  \\\n",
       "#SampleID                                                           \n",
       "11548.A16.91.4    Mouse 91 stool collection 4 of 20  Regular chow   \n",
       "11548.A15.85.11  Mouse 85 stool collection 11 of 20  Regular chow   \n",
       "11548.A15.87.8    Mouse 87 stool collection 8 of 20  Regular chow   \n",
       "11548.A16.92.3    Mouse 92 stool collection 3 of 20  Regular chow   \n",
       "11548.A16.89.6    Mouse 89 stool collection 6 of 20  Regular chow   \n",
       "\n",
       "                 dna_extracted  ... sample_name_orit  sample_type  \\\n",
       "#SampleID                       ...                                 \n",
       "11548.A16.91.4            True  ...         a16.91.4        feces   \n",
       "11548.A15.85.11           True  ...        a15.85.11        feces   \n",
       "11548.A15.87.8            True  ...         a15.87.8        feces   \n",
       "11548.A16.92.3            True  ...         a16.92.3        feces   \n",
       "11548.A16.89.6            True  ...         A16.89.6        feces   \n",
       "\n",
       "                      scientific_name   sex taxon_id     timepoint  \\\n",
       "#SampleID                                                            \n",
       "11548.A16.91.4   mouse gut metagenome  male   410661  not provided   \n",
       "11548.A15.85.11  mouse gut metagenome  male   410661  not provided   \n",
       "11548.A15.87.8   mouse gut metagenome  male   410661  not provided   \n",
       "11548.A16.92.3   mouse gut metagenome  male   410661  not provided   \n",
       "11548.A16.89.6   mouse gut metagenome  male   410661           NaN   \n",
       "\n",
       "                       title     treatment diet_short diet_exp  \n",
       "#SampleID                                                       \n",
       "11548.A16.91.4   10 week IHC  not provided         RC   RC_AIR  \n",
       "11548.A15.85.11  10 week IHC  not provided         RC   RC_AIR  \n",
       "11548.A15.87.8   10 week IHC  not provided         RC   RC_AIR  \n",
       "11548.A16.92.3   10 week IHC  not provided         RC   RC_AIR  \n",
       "11548.A16.89.6   10 week IHC           NaN         RC   RC_AIR  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# import log-ratio data\n",
    "metadata_one = pd.read_csv('../data/Microbiome/longIHC_metadata_cleaned.txt',\n",
    "                           sep='\\t', index_col=0)\n",
    "# import rest of the metadata\n",
    "metadata_two = pd.read_csv('../data/Microbiome/lr-metrics/tempted/tempted-qurro_Muribaculaceae-Akkermansiaceae_LR.tsv',\n",
    "                           sep='\\t', index_col=0)[['Current_Natural_Log_Ratio']]\n",
    "#rename LR column so you know what they are\n",
    "metadata_two.rename(columns={'Current_Natural_Log_Ratio':'Muribaculaceae_Akkermansiaceae'}, inplace=True)\n",
    "# merge the data\n",
    "log_ratio_metdata = pd.concat([metadata_two, metadata_one], axis=1)\n",
    "#drop rows with no value for LR\n",
    "log_ratio_metdata = log_ratio_metdata.dropna(subset=['Muribaculaceae_Akkermansiaceae'])\n",
    "# export in QIIME2 format\n",
    "log_ratio_metdata.index.name = '#SampleID'\n",
    "log_ratio_metdata.to_csv('../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_merged.tsv', sep='\\t')\n",
    "log_ratio_metdata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "log_ratio_metdata[log_ratio_metdata['diet_exp'].isin(['RC_AIR', 'HFD_AIR'])].to_csv('../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_RCairHFDair.tsv', sep='\\t')\n",
    "\n",
    "log_ratio_metdata[log_ratio_metdata['diet_exp'].isin(['RC_AIR', 'HFD_IHC'])].to_csv('../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_RCairHFDihc.tsv', sep='\\t')\n",
    "\n",
    "log_ratio_metdata[log_ratio_metdata['diet_exp'].isin(['HFD_AIR', 'HFD_IHC'])].to_csv('../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_HFDairHFDihc.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_volplot.qzv\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime longitudinal volatility \\\n",
    "    --m-metadata-file ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_merged.tsv \\\n",
    "    --p-state-column host_age \\\n",
    "    --p-individual-id-column host_subject_id \\\n",
    "    --p-default-group-column diet_exp \\\n",
    "    --p-default-metric Muribaculaceae_Akkermansiaceae \\\n",
    "    --o-visualization ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_volplot.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_LME.qzv\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime longitudinal linear-mixed-effects\\\n",
    "    --m-metadata-file ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_merged.tsv \\\n",
    "    --p-state-column host_age \\\n",
    "    --p-individual-id-column host_subject_id \\\n",
    "    --p-group-columns diet_exp \\\n",
    "    --p-metric Muribaculaceae_Akkermansiaceae \\\n",
    "    --o-visualization ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_LME.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_RCairHFDair_LME.qzv\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime longitudinal linear-mixed-effects\\\n",
    "    --m-metadata-file ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_RCairHFDair.tsv \\\n",
    "    --p-state-column host_age \\\n",
    "    --p-individual-id-column host_subject_id \\\n",
    "    --p-group-columns diet_exp \\\n",
    "    --p-metric Muribaculaceae_Akkermansiaceae \\\n",
    "    --o-visualization ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_RCairHFDair_LME.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_RCairHFDihc_LME.qzv\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime longitudinal linear-mixed-effects\\\n",
    "    --m-metadata-file ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_RCairHFDihc.tsv \\\n",
    "    --p-state-column host_age \\\n",
    "    --p-individual-id-column host_subject_id \\\n",
    "    --p-group-columns diet_exp \\\n",
    "    --p-metric Muribaculaceae_Akkermansiaceae \\\n",
    "    --o-visualization ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_RCairHFDihc_LME.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_HFDairHFDihc_LME.qzv\u001b[0m\r\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!qiime longitudinal linear-mixed-effects\\\n",
    "    --m-metadata-file ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_HFDairHFDihc.tsv \\\n",
    "    --p-state-column host_age \\\n",
    "    --p-individual-id-column host_subject_id \\\n",
    "    --p-group-columns diet_exp \\\n",
    "    --p-metric Muribaculaceae_Akkermansiaceae \\\n",
    "    --o-visualization ../data/Microbiome/lr-metrics/tempted/MuribaculaceaeAkkermansiaceae_LR_HFDairHFDihc_LME.qzv"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
